src/packages/cli/src/detach.ts

```diff
@@ -0,0 +1,259 @@
+import psList from "@trufflesuite/ps-list";
+import {
+  existsSync,
+  mkdirSync,
+  rmSync,
+  writeFileSync,
+  readdirSync,
+  readFileSync
+} from "fs";
+import path from "path";
+import { StartArgs } from "./types";
+import { FlavorName } from "@ganache/flavors";
+
+export type DetachedInstance = {
+  instanceName: string;
+  pid: number;
+  startTime: number;
+  host: string;
+  port: number;
+  flavor: FlavorName;
+  cmd: string;
+  version: string;
+};
+
+const dataPath = envPaths(`Ganache/instances`).data;
+if (!existsSync(dataPath)) mkdirSync(dataPath, { recursive: true });
+
+const READY_MESSAGE = "ready";
+
+const START_ERROR = "An error ocurred spawning a detached instance of Ganache:";
+
+/**
+ * Notify that the detached instance has started and is ready to receive requests.
+ */
+export function notifyDetachedInstanceReady() {
+  // in "detach" mode, the parent will wait until the "ready" message is
+  // received before disconnecting from the child process.
+  process.send(READY_MESSAGE);
+}
+
+/**
+ * Attempt to find and remove the instance file for a detached instance.
+ * @param  {number} pid the pid of the detached instance
+ * @returns boolean indicating whether the instance file was cleaned up successfully
+ */
+export function removeDetachedInstanceFile(pid: number): boolean {
+  const instanceFilename = `${dataPath}/${pid}`;
+  if (existsSync(instanceFilename)) {
+    rmSync(instanceFilename);
+    return true;
+  }
+  return false;
+}
+
+/**
+ * Attempts to stop a detached instance with the specified instance name by
+ * sending a SIGTERM signal. Returns a boolean indicating whether the process
+ * was found. If the PID is identified, but the process is not found, any
+ * corresponding instance file will be removed.
+ *
+ * Note: This does not guarantee that the instance actually stops.
+ * @param  {string} instanceName
+ * @returns boolean indicating whether the instance was found.
+ */
+export async function stopDetachedInstance(
+  instanceName: string
+): Promise<boolean> {
+  const instance = await findDetachedInstanceByName(instanceName);
+  if (instance !== undefined) {
+    try {
+      process.kill(instance.pid, "SIGTERM");
+    } catch (err) {
+      // process.kill throws if the process was not found (or was a group process in Windows)
+      return false;
+    } finally {
+      removeDetachedInstanceFile(instance.pid);
+    }
+    return true;
+  }
+  return false;
+}
+
+/**
+ * Start an instance of Ganache in detached mode.
+ * @param  {string[]} argv arguments to be passed to the new instance.
+ * @returns {Promise<DetachedInstance>} resolves to the DetachedInstance once it
+ * is started and ready to receive requests.
+ */
+export async function startDetachedInstance(
+  module: string,
+  args: StartArgs<FlavorName>,
+  version: string
+): Promise<DetachedInstance> {
+  const flavor = args.flavor;
+  const childArgs = createFlatChildArgs(args);
+  const child = fork(module, childArgs, {
```

Comment:
I don't think this will work in some cases, like `ganache --fork`, or a contrived but probably possible with some combination of arguments: `ganache --namespace.someOption=' <- quoted spaces -> '. I think we need to pass the original `argv` args, sans `--detach` to the child process.

Reply:
Fixed